### ISSUE TITLE
LIFO queue strategy / toggle strategy

### ISSUE THREAD

**Issue by dt-rush** [brianc/node-postgres#1749]
In certain circumstances, such as when the DB is overloaded and requests are waiting in the queue for long enough to approach a timeout limit, switching to a LIFO strategy temporarily actually can ensure better performance.

**Comment by vitaly-t** [comment]
Great, except it has no bearing on this library.

**Comment by dt-rush** [comment]
If you mean "this is scoped to `node-pg-pool`", I opened this in `node-postgres` because if we are going to implement this in `node-pg-pool`, it will need to be available as an option / documented here as well. 

If you mean, on the other hand, "this library doesn't care about FIFO/LIFO", you're simply saying that "this library doesn't care about flexibility and performance". I'm sure this isn't correct. I'll be writing a PR for `node-pg-pool`, and when it's implemented, we can review whether a feature of the underlying pooled connector would be relevant to make available to a coupled module such as `node-postgres`.

**Comment by charmander** [comment]
LIFO where? Do you mean sending the newest query to the server instead of the oldest one?

**Comment by dt-rush** [comment]
@charmander exactly, the idea being that when overloaded, it's better to serve requests which have the shortest wait time rather than spending cycles processing those which are more likely to be close to any timeouts defined at a prior level of the serving infrastructure
